src/parser/mod.rs

```diff
@@ -6484,11 +6496,99 @@ impl<'a> Parser<'a> {
         }
     }
 
-    pub fn parse_sql_option(&mut self) -> Result<SqlOption, ParserError> {
+    pub fn parse_key_value(&mut self) -> Result<(Ident, Expr), ParserError> {
         let name = self.parse_identifier(false)?;
         self.expect_token(&Token::Eq)?;
         let value = self.parse_expr()?;
-        Ok(SqlOption { name, value })
+
+        Ok((name, value))
+    }
+
+    pub fn parse_sql_option(&mut self) -> Result<SqlOption, ParserError> {
+        let next_token = self.peek_token();
+        let is_mssql = dialect_of!(self is MsSqlDialect|GenericDialect);
+
+        let Token::Word(w) = next_token.token else {
+            let (name, value) = self.parse_key_value()?;
+            return Ok(SqlOption::KeyValue { name, value });
+        };
```

Comment:
Rewrote the match arm to match the keyword in the match arm condition. It was "needed" since not all options are `Token::Word`, but now the last match arm will handle both those cases and the ones where they _are_ `Token::Word` but not any of the special ones.